src/huggingface_hub/repository.py

```diff
@@ -1032,48 +1032,70 @@ def git_checkout(self, revision: str, create_branch_ok: Optional[bool] = False):
                 except subprocess.CalledProcessError as exc:
                     raise EnvironmentError(exc.stderr)
 
-    def tag_exists(self, tag_name: str) -> bool:
+    def tag_exists(self, tag_name: str, remote: Optional[str] = None) -> bool:
```

Comment:
I'm adding a way to check if a tag exists remotely, on top of checking if the tag exists locally